revocation/revocation.go

```diff
@@ -33,10 +35,31 @@ type Revocation interface {
 	Validate(certChain []*x509.Certificate, signingTime time.Time) ([]*result.CertRevocationResult, error)
 }
 
+// ValidateContextOptions provides configuration options for revocation checks
+type ValidateContextOptions struct {
+	// CertChain denotes the certificate chain whose revocation status is
+	// been validated. REQUIRED.
+	CertChain []*x509.Certificate
+
+	// AuthenticSigningTime denotes the authentic signing time of the signature.
+	// It is solely used under signing scheme `notary.x509.signingAuthority`.
+	// OPTIONAL.
+	AuthenticSigningTime time.Time
+}
+
+// ContextRevocation is an interface that provides revocation checking with
+// context
+type ContextRevocation interface {
+	// ValidateContext checks the revocation status given caller provided options
+	// and returns an array of CertRevocationResults that contain the results
+	// and any errors that are encountered during the process
+	ValidateContext(ctx context.Context, validateContextOpts ValidateContextOptions) ([]*result.CertRevocationResult, error)
+}
+
 // revocation is an internal struct used for revocation checking
 type revocation struct {
 	httpClient       *http.Client
-	certChainPurpose ocsp.Purpose
+	certChainPurpose x509.ExtKeyUsage
 }
 
 // New constructs a revocation object for code signing certificate chain
```

Comment:
Should you mark `New` and `Revocation` as deprecated?

Reply:
I think we could do it in the next stable release. `notation-go` is still using them. The CRL PR will refactor the `revocation.go` file later on, we can deprecate them in that PR.

Reply:
Updated.